### ISSUE TITLE
iOS 10 : Crashes in RCNConfigDBManager and FIRIAMActionURLFollower

### ISSUE THREAD

**Issue by Max-3-7** [firebase/firebase-ios-sdk#5296]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.4
  * Firebase SDK version: 6.21.0
  * Firebase Component: FirebaseRemoteConfig
  * Component version: 4.4.7
  * Installation method: CocoaPods

#### Steps to reproduce:

We've recently updated from version 6.17.0 to 6.21.0 and now we are getting crashes on iOS 10 (only). It seems to be related to RemoteConfig. 
We didn't change anything in the code that uses remote config between these two releases.

The crashes are :
```
Plantage : com.apple.main-thread
EXC_BAD_ACCESS KERN_PROTECTION_FAILURE 0x00000000382f4000
```

Here are some stack traces :
<details>
<summary>Stacktrace</summary>
<pre>
FIRIAMActionURLFollower.m line 77
__44+[FIRIAMActionURLFollower actionURLFollower]_block_invoke
```
com.apple.root.default-qos
0  ???                            0x1c3de554 (Manquant)
1  ???                            0x1c2eea9d (Manquant)
2  ???                            0x1c2eeb9b (Manquant)
3  ???                            0x1c2ecb39 (Manquant)
4  ***                           0x40eae7 __44+[FIRIAMActionURLFollower actionURLFollower]_block_invoke + 77 (FIRIAMActionURLFollower.m:77)
5  ???                            0x1c2de253 (Manquant)
6  ***                           0x40e7f3 +[FIRIAMActionURLFollower actionURLFollower] + 84 (once.h:84)
7  ***                           0x429af9 -[FIRIAMRuntimeManager internalStartRuntimeWithSDKSettings:] + 327 (FIRIAMRuntimeManager.m:327)
8  ???                            0x1c2dd797 (Manquant)
9  ???                            0x1c2eab1d (Manquant)
10 ???                            0x1c2ec1b5 (Manquant)
11 ???                            0x1c2ec00f (Manquant)
12 ???                            0x1c49287d (Manquant)
13 ???                            0x1c49245c (Manquant)
```

RCNConfigDBManager.m line 450
-[RCNConfigDBManager insertMainTableWithValues:fromSource:]
```
com.google.GoogleConfigService.database
0  ???                            0x1c3ca2fc (Manquant)
1  ???                            0x1d0b6cd1 (Manquant)
2  ???                            0x1d0bf23b (Manquant)
3  ???                            0x1d0b67fd (Manquant)
4  ???                            0x1d0a7eb7 (Manquant)
5  ???                            0x1d0818fb (Manquant)
6  ???                            0x1d0a2e17 (Manquant)
7  ???                            0x1d0965ab (Manquant)
8  ***                           0x169f717 -[RCNConfigDBManager insertMainTableWithValues:fromSource:] + 450 (RCNConfigDBManager.m:450)
9  ***                           0x169f499 __77-[RCNConfigDBManager insertMainTableWithValues:fromSource:completionHandler:]_block_invoke + 401 (RCNConfigDBManager.m:401)
10 ???                            0x1c2dd797 (Manquant)
11 ???                            0x1c2ea59d (Manquant)
12 ???                            0x1c2e0b71 (Manquant)
13 ???                            0x1c2eaa1f (Manquant)
14 ???                            0x1c2ec1b5 (Manquant)
15 ???                            0x1c2ec00f (Manquant)
16 ???                            0x1c49287d (Manquant)
17 ???                            0x1c49245c (Manquant)
```

RCNConfigDBManager.m line 979
-[RCNConfigDBManager executeQuery:withParams:]
```
com.google.GoogleConfigService.database
0  ???                            0x1d5762fc (Manquant)
1  ???                            0x1e262cd1 (Manquant)
2  ???                            0x1e26b23b (Manquant)
3  ???                            0x1e2627fd (Manquant)
4  ???                            0x1e253eb7 (Manquant)
5  ???                            0x1e22d8fb (Manquant)
6  ???                            0x1e24ee17 (Manquant)
7  ???                            0x1e2425ab (Manquant)
8  ***                           0x1627a23 -[RCNConfigDBManager executeQuery:withParams:] + 979 (RCNConfigDBManager.m:979)
9  ***                           0x16276cf __89-[RCNConfigDBManager deleteRecordFromMainTableWithNamespace:bundleIdentifier:fromSource:]_block_invoke + 920 (RCNConfigDBManager.m:920)
10 ???                            0x1d489797 (Manquant)
11 ???                            0x1d49659d (Manquant)
12 ???                            0x1d48cb71 (Manquant)
13 ???                            0x1d496a1f (Manquant)
14 ???                            0x1d4981b5 (Manquant)
15 ???                            0x1d49800f (Manquant)
16 ???                            0x1d63e87d (Manquant)
17 ???                            0x1d63e45c (Manquant)
```

GULLogger.m line 173
__GULLogBasic_block_invoke
```
GULLoggingClientQueue
0  ???                            0x1d43fcc2 (Manquant)
1  ???                            0x1d366375 (Manquant)
2  ???                            0x1d440c79 (Manquant)
3  ???                            0x1d440b8d (Manquant)
4  ???                            0x1d2a8219 (Manquant)
5  GoogleUtilities                0x271bbb9 __GULLogBasic_block_invoke + 173 (GULLogger.m:173)
6  ???                            0x1d26d797 (Manquant)
7  ???                            0x1d27a59d (Manquant)
8  ???                            0x1d270b71 (Manquant)
9  ???                            0x1d27c1b5 (Manquant)
10 ???                            0x1d27c00f (Manquant)
11 ???                            0x1d42287d (Manquant)
12 ???                            0x1d42245c (Manquant)
```

RCNUserDefaultsManager.m line 199
-[RCNUserDefaultsManager instanceUserDefaults]
```
com.google.GoogleConfigService.FIRRemoteConfig
0  ???                            0x1afdd920 (Manquant)
1  ???                            0x1b092cc9 (Manquant)
2  ???                            0x1aa9ed95 (Manquant)
3  ???                            0x1aa9eb9f (Manquant)
4  ???                            0x1aa9eb65 (Manquant)
5  ???                            0x1aaa51af (Manquant)
6  ???                            0x1b7a442b (Manquant)
7  ???                            0x1b7c232d (Manquant)
8  ???                            0x1b0bc04f (Manquant)
9  ???                            0x1b7c2273 (Manquant)
10 ???                            0x1b7c2ab7 (Manquant)
11 ???                            0x1b84858d (Manquant)
12 ???                            0x1b7c2a77 (Manquant)
13 ???                            0x1b0a7305 (Manquant)
14 ???                            0x1b0a7327 (Manquant)
15 ???                            0x1b7c29db (Manquant)
16 ???                            0x1b7c25a5 (Manquant)
17 ???                            0x1b7c27a1 (Manquant)
18 ???                            0x1b7c2ec9 (Manquant)
19 ???                            0x1b7c17d5 (Manquant)
20 ???                            0x1b841407 (Manquant)
21 ???                            0x1b846d09 (Manquant)
22 ???                            0x1b7c41df (Manquant)
23 ???                            0x1b7c3b21 (Manquant)
24 ???                            0x1b7c40a1 (Manquant)
25 ???                            0x1b846c6d (Manquant)
26 ???                            0x1b849803 (Manquant)
27 ???                            0x1c071e11 (Manquant)
28 ???                            0x1c0713c9 (Manquant)
29 ***                           0x15e507d -[RCNUserDefaultsManager instanceUserDefaults] + 199 (RCNUserDefaultsManager.m:199)
30 ***                           0x15e4c81 -[RCNUserDefaultsManager lastFetchTime] + 124 (RCNUserDefaultsManager.m:124)
31 ***                           0x15e04a7 -[RCNConfigSettings hasMinimumFetchIntervalElapsed:] + 445 (RCNConfigSettings.m:445)
32 ***                           0x15e21d1 __70-[RCNConfigFetch fetchConfigWithExpirationDuration:completionHandler:]_block_invoke + 144 (RCNFetch.m:144)
33 ???                            0x1aedd797 (Manquant)
34 ???                            0x1aeea59d (Manquant)
35 ???                            0x1aee0b71 (Manquant)
36 ???                            0x1aeeaa1f (Manquant)
37 ???                            0x1aeec1b5 (Manquant)
38 ???                            0x1aeec00f (Manquant)
39 ???                            0x1b09287d (Manquant)
40 ???                            0x1b09245c (Manquant)
```

ANSUploader.m line 107
-[ANSUploader supportsBackgroundRequests]
```
io.answers.EventQueue :: NSOperation 0x16e42290 (QOS: BACKGROUND)
0  ???                            0x1b4ecdde (Manquant)
1  ***                           0x2e22fb -[ANSUploader supportsBackgroundRequests] + 107 (ANSUploader.m:107)
2  ***                           0x2e23d1 __70-[ANSUploader attemptToReconnectBackgroundSessionWithCompletionBlock:]_block_invoke + 113 (ANSUploader.m:113)
3  Foundation                     0x1cb7db01 <redacted> + 8
4  Foundation                     0x1cad3aef <redacted> + 146
5  Foundation                     0x1cac5fe7 <redacted> + 770
6  Foundation                     0x1cb7fd39 <redacted> + 190
7  ???                            0x1b93259d (Manquant)
8  ???                            0x1b928b71 (Manquant)
9  ???                            0x1b9341b5 (Manquant)
10 ???                            0x1b93400f (Manquant)
11 ???                            0x1bada87d (Manquant)
12 ???                            0x1bada45c (Manquant)
```

RCNConfigDBManager.m line 808
-[RCNConfigDBManager loadInternalMetadataTable]
```
com.google.perf.FPREventsQueue
0  ???                            0x1b226554 (Manquant)
1  ???                            0x1b136a9d (Manquant)
2  ???                            0x1b136b9b (Manquant)
3  ???                            0x1b134b39 (Manquant)
4  ***                           0x15e6d69 -[RCNConfigDBManager loadInternalMetadataTable] + 808 (RCNConfigDBManager.m:808)
5  ***                           0x15e8db7 -[RCNConfigSettings initWithDatabaseManager:namespace:firebaseAppName:googleAppID:] + 98 (RCNConfigSettings.m:98)
6  ***                           0x15e0321 -[FIRRemoteConfig initWithAppName:FIROptions:namespace:DBManager:configContent:analytics:] + 160 (FIRRemoteConfig.m:160)
7  ***                           0x15e2d61 -[FIRRemoteConfigComponent remoteConfigForNamespace:] + 73 (FIRRemoteConfigComponent.m:73)
8  ***                           0x15e00cf +[FIRRemoteConfig remoteConfigWithFIRNamespace:app:] + 117 (FIRRemoteConfig.m:117)
9  ***                           0x15866b5 __38+[FPRRemoteConfigFlags sharedInstance]_block_invoke + 23258805
10 ???                            0x1b125783 (Manquant)
11 ???                            0x1b126253 (Manquant)
12 ***                           0x1586661 +[FPRRemoteConfigFlags sharedInstance] + 23258721
13 ***                           0x1585869 -[FPRConfigurations setupRemoteConfigFlags] + 23255145
14 ***                           0x15857af -[FPRConfigurations update] + 23254959
15 ***                           0x1588fa1 __42-[FPRClient startWithConfiguration:error:]_block_invoke + 23269281
16 ???                            0x1b125797 (Manquant)
17 ???                            0x1b132339 (Manquant)
18 ???                            0x1b128b71 (Manquant)
19 ???                            0x1b132a1f (Manquant)
20 ???                            0x1b1341b5 (Manquant)
21 ???                            0x1b13400f (Manquant)
22 ???                            0x1b2da87d (Manquant)
23 ???                            0x1b2da45c (Manquant)
```
</pre>
</details>

#### Relevant Code:

Basically, from `func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool`

We call : 
- FirebaseApp.configure()
- Performance.sharedInstance().isInstrumentationEnabled = false
- RemoteConfig.remoteConfig().setDefaults(fromPlist: "xxx")
- (void)fetchWithExpirationDuration:(NSTimeInterval)expirationDuration
                             completionHandler:(nullable FIRRemoteConfigFetchCompletion) completionHandler;

The crashes occur at launch.

Thanks for your help !

**Comment by ppamorim** [comment]
This exact same issue is happening to me.

**Comment by morganchen12** [comment]
Do either of you have any sqlite code in your codebases?

**Comment by ppamorim** [comment]
@morganchen12 No.

**Comment by Max-3-7** [comment]
@morganchen12 No!

**Comment by morganchen12** [comment]
Can you both share fully symbolicated stack traces of the crash?

**Comment by paulb777** [comment]
The second backtrace looks like it might be related to an interaction between FirebaseAnalytics and FirebaseRemoteConfig.

**Comment by ppamorim** [comment]
@morganchen12 The crash happened on the client side. I am not sure how to get this on Firebase. The report didn't appear yet on Xcode.

**Comment by morganchen12** [comment]
Based on the second stack trace it looks like Analytics/RC are trying to write to sqlite at the same time, which ordinarily resolves without issue. Since you mentioned this crash occurs at launch, it'd be really helpful to know if this was the result of an exception in the sqlite calls or if the system terminated the apps at launch because `didFinishLaunchingWithOptions` was taking too long to return.

**Comment by lemonlab** [comment]
I'm having the same error since i updated the SDK, on iOS 10 only.
Here is the full stack trace .

**com.apple.main-thread**
```
Crashed: com.apple.main-thread
EXC_BAD_ACCESS KERN_PROTECTION_FAILURE 0x00000000383a8000
```

**com.apple.root.default-qos**
```
0  ???                            0x1b256554 (Missing)
[...]
4  MyApp                           0x477ae7 __44+[FIRIAMActionURLFollower actionURLFollower]_block_invoke + 77 (FIRIAMActionURLFollower.m:77)
5  ???                            0x1b156253 (Missing)
6  MyApp                           0x4777f3 +[FIRIAMActionURLFollower actionURLFollower] + 84 (once.h:84)
7  MyApp                           0x492af9 -[FIRIAMRuntimeManager internalStartRuntimeWithSDKSettings:] + 327 (FIRIAMRuntimeManager.m:327)
8  ???                            0x1b155797 (Missing)
[...]
```

**APMMonitorQueue**
```
0  ???                            0x1b254568 (Missing)
[...]
7  MyApp                           0x16f9a09 -[APMUserDefaults scheduleSynchronize] + 23792137
8  MyApp                           0x16f921b -[APMUserDefaults setObject:forKey:] + 23790107
9  MyApp                           0x16cfb4d -[APMMonitor migrateMonitorDataOnMonitorQueue] + 23620429
10 MyApp                           0x16cdc4f __53-[APMMonitor initWithLogLevel:logger:libraryVersion:]_block_invoke + 23612495
11 ???                            0x1b155797 (Missing)
[...]
```

**APMAnalyticsQueue**
```
0  ???                            0x1b254568 (Missing)
[...]
7  MyApp                           0x1697af3 -[APMEManager init] + 23390963
8  MyApp                           0x1697973 __29+[APMEManager sharedInstance]_block_invoke + 23390579
9  ???                            0x1b156253 (Missing)
10 MyApp                           0x1697941 +[APMEManager sharedInstance] + 23390529
11 MyApp                           0x16b7db3 +[APMMeasurement initializeSharedInstanceWithAppID:isAnalyticsCollectionEnabled:isAnalyticsCollectionDeactivated:safelistedEvents:] + 23522739
12 MyApp                           0x16a1d5d __46+[APMAnalytics startWithAppID:origin:options:]_block_invoke_2 + 23432541
13 ???                            0x1b155797 (Missing)
[...]
```

**com.google.GoogleConfigService.database**
```
0  ???                            0x1b2548c4 (Missing)
[...]
8  MyApp                           0x1662717 -[RCNConfigDBManager insertMainTableWithValues:fromSource:] + 450 (RCNConfigDBManager.m:450)
9  MyApp                           0x1662499 __77-[RCNConfigDBManager insertMainTableWithValues:fromSource:completionHandler:]_block_invoke + 401 (RCNConfigDBManager.m:401)
10 ???                            0x1b155797 (Missing)
[...]
```

**com.apple.NSURLConnectionLoader**
```
0  ???                            0x1b240900 (Missing)
[...]
10 ???                            0x1b30a468 (Missing)
```

**com.apple.uikit.eventfetch-thread**
```
0  ???                            0x1b240900 (Missing)
[...]
12 ???                            0x1b30a468 (Missing)
```

**com.apple.uikit.eventfetch-thread**
```
0  ???                            0x1b240900 (Missing)
[...]
12 ???                            0x1b30a468 (Missing)
```

**com.twitter.crashlytics.ios.binary-images**
```
0  ???                            0x1b254a48 (Missing)
[...]
11 MyApp                           0x29861d __CLSBinaryImageChanged_block_invoke + 2422301
12 ???                            0x1b155797 (Missing)
[...]
```

**com.twitter.crashlytics.ios.MachExceptionServer**
```
0  MyApp                           0x2abe55 CLSProcessRecordAllThreads + 2502229
1  MyApp                           0x2abe55 CLSProcessRecordAllThreads + 2502229
2  MyApp                           0x2ac0b3 CLSProcessRecordAllThreads + 2502835
3  MyApp                           0x29f70d CLSHandler + 26 (CLSHandler.m:26)
4  MyApp                           0x29bb75 CLSMachExceptionServer + 2435957
5  ???                            0x1b30c93b (Missing)
[...]
```

**CCTLogWriter**
```
0  ???                            0x1badc960 (Missing)
[...]
7  MyApp                           0x164c421 -[CCTUIKitAXUtils initPrivate] + 23082017
8  MyApp                           0x164c35f __33+[CCTUIKitAXUtils sharedInstance]_block_invoke + 23081823
9  ???                            0x1b156253 (Missing)
10 MyApp                           0x164c32d +[CCTUIKitAXUtils sharedInstance] + 23081773
11 MyApp                           0x1647c75 __54-[CCTLogWriter ensureAccessibilityCacheForWriteBlock:]_block_invoke + 23063669
12 ???                            0x1b155797 (Missing)
[...]
```

Thank you for your help !

**Comment by Max-3-7** [comment]
Here's the full stack trace from Crashlytics :

RCNConfigDBManager.m line 450
-[RCNConfigDBManager insertMainTableWithValues:fromSource:] 

Plantage : com.apple.main-thread
```
EXC_BAD_ACCESS KERN_PROTECTION_FAILURE 0x0000000038b50000
Crashed: com.apple.main-thread
0  ???                            0x38b50000 (Manquant)
```

com.google.GoogleConfigService.database
```
com.google.GoogleConfigService.database
0  ???                            0x1b9eada0 (Manquant)
1  ???                            0x1c6df6a9 (Manquant)
2  ???                            0x1c6c8645 (Manquant)
3  ???                            0x1c6c8183 (Manquant)
4  ???                            0x1c6a1923 (Manquant)
5  ???                            0x1c6c2e17 (Manquant)
6  ???                            0x1c6b65ab (Manquant)
7  App                           0x1694717 -[RCNConfigDBManager insertMainTableWithValues:fromSource:] + 450 (RCNConfigDBManager.m:450)
8  App                           0x1694499 __77-[RCNConfigDBManager insertMainTableWithValues:fromSource:completionHandler:]_block_invoke + 401 (RCNConfigDBManager.m:401)
9  ???                            0x1b8fd797 (Manquant)
10 ???                            0x1b90a59d (Manquant)
11 ???                            0x1b900b71 (Manquant)
12 ???                            0x1b90aa1f (Manquant)
13 ???                            0x1b90c1b5 (Manquant)
14 ???                            0x1b90c00f (Manquant)
15 ???                            0x1bab287d (Manquant)
16 ???                            0x1bab245c (Manquant)
```

GULLoggingClientQueue
```
GULLoggingClientQueue
0  GoogleUtilities                0x27913a4 __destroy_helper_block_e4_20s + 465 (GULNetworkURLSession.m:465)
1  ???                            0x1b947bb9 (Manquant)
2  ???                            0x1b90a59d (Manquant)
3  ???                            0x1b900b71 (Manquant)
4  ???                            0x1b90c1b5 (Manquant)
5  ???                            0x1b90c00f (Manquant)
6  ???                            0x1bab287d (Manquant)
7  ???                            0x1bab245c (Manquant)
```

Thread 1
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab28eb (Manquant)
2  ???                            0x1bab245c (Manquant)
```

Thread 2
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab28eb (Manquant)
2  ???                            0x1bab245c (Manquant)
```

com.apple.uikit.eventfetch-thread
```
com.apple.uikit.eventfetch-thread
0  ???                            0x1b9e8900 (Manquant)
1  ???                            0x1b9e86e1 (Manquant)
2  ???                            0x1c1ebbe3 (Manquant)
3  ???                            0x1c1ea065 (Manquant)
4  ???                            0x1c13d1af (Manquant)
5  ???                            0x1c13cfd1 (Manquant)
6  ???                            0x1ca91af5 (Manquant)
7  ???                            0x1caae76d (Manquant)
8  ???                            0x21dc6ad9 (Manquant)
9  ???                            0x1cb758eb (Manquant)
10 ???                            0x1bab493b (Manquant)
11 ???                            0x1bab485d (Manquant)
12 ???                            0x1bab2468 (Manquant)
```

APMAnalyticsQueue
```
APMAnalyticsQueue
0  ???                            0x1b9fc568 (Manquant)
1  ???                            0x1bab83bf (Manquant)
2  ???                            0x1b4bedd5 (Manquant)
3  ???                            0x1b4bed07 (Manquant)
4  ???                            0x1b4beb65 (Manquant)
5  ???                            0x1b4c51af (Manquant)
6  GoogleUtilities                0x278e4b7 -[GULNetwork initWithReachabilityHost:] + 73 (GULNetwork.m:73)
7  App                           0x16ce451 -[APMETaskManager init] + 23266385
8  App                           0x16c9b03 -[APMEManager init] + 23247619
9  App                           0x16c9973 __29+[APMEManager sharedInstance]_block_invoke + 23247219
10 ???                            0x1b8fe253 (Manquant)
11 App                           0x16c9941 +[APMEManager sharedInstance] + 23247169
12 App                           0x16e9db3 +[APMMeasurement initializeSharedInstanceWithAppID:isAnalyticsCollectionEnabled:isAnalyticsCollectionDeactivated:safelistedEvents:] + 23379379
13 App                           0x16d3d5d __46+[APMAnalytics startWithAppID:origin:options:]_block_invoke_2 + 23289181
14 ???                            0x1b8fd797 (Manquant)
15 ???                            0x1b90a59d (Manquant)
16 ???                            0x1b900b71 (Manquant)
17 ???                            0x1b90c1b5 (Manquant)
18 ???                            0x1b90c00f (Manquant)
19 ???                            0x1bab287d (Manquant)
20 ???                            0x1bab245c (Manquant)
```

com.apple.root.default-qos
```
com.apple.root.default-qos
0  ???                            0x1b9fe554 (Manquant)
1  ???                            0x1b90ea9d (Manquant)
2  ???                            0x1b90eb9b (Manquant)
3  ???                            0x1b90cb39 (Manquant)
4  App                           0x4a9ae7 __44+[FIRIAMActionURLFollower actionURLFollower]_block_invoke + 77 (FIRIAMActionURLFollower.m:77)
5  ???                            0x1b8fe253 (Manquant)
6  App                           0x4a97f3 +[FIRIAMActionURLFollower actionURLFollower] + 84 (once.h:84)
7  App                           0x4c4af9 -[FIRIAMRuntimeManager internalStartRuntimeWithSDKSettings:] + 327 (FIRIAMRuntimeManager.m:327)
8  ???                            0x1b8fd797 (Manquant)
9  ???                            0x1b90ab1d (Manquant)
10 ???                            0x1b90c1b5 (Manquant)
11 ???                            0x1b90c00f (Manquant)
12 ???                            0x1bab287d (Manquant)
13 ???                            0x1bab245c (Manquant)
```

Thread 3
```
Thread
0  ???                            0x1b9fde7c (Manquant)
1  ???                            0x1b9520e9 (Manquant)
2  ???                            0x1b48f92b (Manquant)
3  ???                            0x200a93c5 (Manquant)
4  ???                            0x200a9207 (Manquant)
5  ???                            0x200a9121 (Manquant)
6  ???                            0x200aa197 (Manquant)
7  ???                            0x200aa0ed (Manquant)
8  ???                            0x200aa32f (Manquant)
9  ???                            0x1bab493b (Manquant)
10 ???                            0x1bab485d (Manquant)
11 ???                            0x1bab2468 (Manquant)
```

CCTLogWriter
```
CCTLogWriter
0  ???                            0x1b9fd954 (Manquant)
1  ???                            0x1bab2f77 (Manquant)
2  ???                            0x1b4bedb1 (Manquant)
3  ???                            0x1b4bebbb (Manquant)
4  ???                            0x1b4beb65 (Manquant)
5  ???                            0x1b4c51af (Manquant)
6  App                           0x1679c75 __54-[CCTLogWriter ensureAccessibilityCacheForWriteBlock:]_block_invoke + 22920309
7  ???                            0x1b8fd797 (Manquant)
8  ???                            0x1b90a59d (Manquant)
9  ???                            0x1b900b71 (Manquant)
10 ???                            0x1b90c1b5 (Manquant)
11 ???                            0x1b90c00f (Manquant)
12 ???                            0x1bab287d (Manquant)
13 ???                            0x1bab245c (Manquant)
```

com.apple.NSURLConnectionLoader
```
com.apple.NSURLConnectionLoader
0  ???                            0x1b9e8900 (Manquant)
1  ???                            0x1b9e86e1 (Manquant)
2  ???                            0x1c1ebbe3 (Manquant)
3  ???                            0x1c1ea065 (Manquant)
4  ???                            0x1c13d1af (Manquant)
5  ???                            0x1c13cfd1 (Manquant)
6  ???                            0x1c81a393 (Manquant)
7  ???                            0x1cb758eb (Manquant)
8  ???                            0x1bab493b (Manquant)
9  ???                            0x1bab485d (Manquant)
10 ???                            0x1bab2468 (Manquant)
```

Thread 4
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab28eb (Manquant)
2  ???                            0x1bab245c (Manquant)
com.twitter.crashlytics.ios.MachExceptionServer
com.twitter.crashlytics.ios.MachExceptionServer
0  App                           0x2dde55 CLSProcessRecordAllThreads + 2358869
1  App                           0x2dde55 CLSProcessRecordAllThreads + 2358869
2  App                           0x2de0b3 CLSProcessRecordAllThreads + 2359475
3  App                           0x2d170d CLSHandler + 26 (CLSHandler.m:26)
4  App                           0x2cdb75 CLSMachExceptionServer + 2292597
5  ???                            0x1bab493b (Manquant)
6  ???                            0x1bab485d (Manquant)
7  ???                            0x1bab2468 (Manquant)
```

com.twitter.crashlytics.ios.binary-images
```
com.twitter.crashlytics.ios.binary-images
0  ???                            0x1c2211e8 (Manquant)
1  ???                            0x1c220dd7 (Manquant)
2  ???                            0x1c13e841 (Manquant)
3  ???                            0x1c1be56f (Manquant)
4  ???                            0x1c1bec91 (Manquant)
5  ???                            0x1c1bf09d (Manquant)
6  ???                            0x1c137a5d (Manquant)
7  ???                            0x1c15c045 (Manquant)
8  ???                            0x1c167413 (Manquant)
9  ???                            0x1c13ed8d (Manquant)
10 ???                            0x1c1666e1 (Manquant)
11 ???                            0x1c172835 (Manquant)
12 ???                            0x1ca8c6f5 (Manquant)
13 ???                            0x1ca8c68f (Manquant)
14 ???                            0x1ca8c657 (Manquant)
15 App                           0x2ca61d __CLSBinaryImageChanged_block_invoke + 2278941
16 ???                            0x1b8fd797 (Manquant)
17 ???                            0x1b90a59d (Manquant)
18 ???                            0x1b900b71 (Manquant)
19 ???                            0x1b90aa1f (Manquant)
20 ???                            0x1b90c1b5 (Manquant)
21 ???                            0x1b90c00f (Manquant)
22 ???                           0x1bab287d (Manquant)
23 ???                            0x1bab245c (Manquant)
```

Thread 5
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab2745 (Manquant)
2  ???                            0x1bab245c (Manquant)
```

Thread 6
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab28eb (Manquant)
2  ???                            0x1bab245c (Manquant)
```

Thread 7
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab2745 (Manquant)
2  ???                            0x1bab245c (Manquant)
```

Thread 8
```
Thread
0  ???                            0x1b9fe73c (Manquant)
1  ???                            0x1bab28eb (Manquant)
2  ???                            0x1bab245c (Manquant)
```

It crashes at many places in RCNConfigDBManager.m (see my first post) but this one is the most frequent.

**Comment by ppamorim** [comment]
@Max-3-7 Could you please start sending this externally to keep the issue clean?

**Comment by morganchen12** [comment]
Does the issue go away if you selectively downgrade RemoteConfig (but keep Analytics at the latest version)?

**Comment by Max-3-7** [comment]
I can't tell, I'm not able to reproduce on an iOS 10 simulator

**Comment by cafielo** [comment]
I'm having the same error since i updated the SDK 6.21.0, on iOS 10 only.
Here is the stack trace .

<details>
    <summary>Stack Trace</summary>


> Crashed: com.apple.main-thread
> EXC_BAD_ACCESS KERN_PROTECTION_FAILURE 0x000000003a808000
> 

```
Crashed: com.apple.main-thread
0  ???                            0x3a808000 (Missing)
1  ???                            0x2900 (Missing)
```


> com.google.GoogleConfigService.database

```
com.google.GoogleConfigService.database
0  libsystem_kernel.dylib         0x1d6b62fc fsync + 8
1  libsqlite3.dylib               0x1e3a2cd1 (null) + 44054
2  libsqlite3.dylib               0x1e3ab29b (null) + 2170
3  libsqlite3.dylib               0x1e3a27fd (null) + 42818
4  libsqlite3.dylib               0x1e393eb7 (null) + 16492
5  libsqlite3.dylib               0x1e36d8fb (null) + 2524
6  libsqlite3.dylib               0x1e38ee17 (null) + 51750
7  libsqlite3.dylib               0x1e3825ab sqlite3_step + 442
8  Alarmy                         0x8263ab -[RCNConfigDBManager insertMainTableWithValues:fromSource:] + 450 (RCNConfigDBManager.m:450)
9  Alarmy                         0x82612d __77-[RCNConfigDBManager insertMainTableWithValues:fromSource:completionHandler:]_block_invoke + 401 (RCNConfigDBManager.m:401)
10 libdispatch.dylib              0x1d5c9797 _dispatch_call_block_and_release + 10
11 libdispatch.dylib              0x1d5d659d _dispatch_queue_serial_drain + 854
12 libdispatch.dylib              0x1d5ccb71 _dispatch_queue_invoke + 886
13 libdispatch.dylib              0x1d5d6a1f _dispatch_queue_override_invoke + 282
14 libdispatch.dylib              0x1d5d81b5 _dispatch_root_queue_drain + 326
15 libdispatch.dylib              0x1d5d800f _dispatch_worker_thread3 + 106
16 libsystem_pthread.dylib        0x1d77e87d _pthread_wqthread + 1040
17 libsystem_pthread.dylib        0x1d77e45c start_wqthread + 8
```

</details>


Thanks 🙏

**Comment by dmandar** [comment]
@Cafielo what last previous version worked for you?

**Comment by cafielo** [comment]
@dmandar 
Our last sdk version was 6.19.0, and it worked well.

**Comment by ppamorim** [comment]
@dmandar For me too. I used 6.19.0 too and we had a regression (this crash) in this current version.

**Comment by paulb777** [comment]
Hmm, RemoteConfig did not change from 6.19.0 to 6.21.0.  The only changing code I see from the backtraces is from APMETaskManager in FirebaseAnalytics, so we'll take a deeper look there.

**Comment by baolocdo** [comment]
Would you mind providing your podfile? Do you have AdMob SDK? Also do you call -[FIRAnalytics appInstanceID] at all?

**Comment by Max-3-7** [comment]
We don't call -[FIRAnalytics appInstanceID] and we don't have AdMob SDK.

<details>
<summary>Podfile</summary>
<pre>
  pod 'Firebase/Core'
  pod 'Firebase/Firestore'
  pod 'Firebase/Database'
  pod 'Firebase/Auth'
  pod 'FirebaseUI/Auth'
  pod 'FirebaseUI/Email'
  pod 'FirebaseUI/Facebook'
  pod 'Firebase/DynamicLinks'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'
  pod 'Firebase/Messaging'
  pod 'Firebase/InAppMessaging'
  pod 'Fabric'
  pod 'Crashlytics'
  pod 'Firebase/Performance'
  pod 'Firebase/MLVision'
  pod 'Firebase/MLVisionFaceModel'
  pod 'Firebase/MLVisionTextModel'
  pod 'Firebase/Functions'
</pre>
</details>

Let me know if you need something else!

**Comment by paulb777** [comment]
There might be an Apple bug building applications with Xcode 11.4 for 32 bit devices. See the discussion at #5233.

Can you confirm that all crashes come from 32bit devices?  Is building with Xcode 11.3.1 a workaround?

**Comment by jordanebelanger** [comment]
I am also seeing what I think is the same issue. I currently have ~20, 1 user 1 event, crashes related to `APMSqliteStore` usually when its calling `-[APMSqliteStore executeSQLStatement:error:]`

They are all happening on iOS 10 only as well.

**Comment by ppamorim** [comment]
Just additional information. We received a crash report of an iOS 13 device and it's pointing to the same crash report above.

**Comment by paulb777** [comment]
From the [Xcode 11.5 Beta release notes](https://developer.apple.com/documentation/xcode_release_notes/xcode_11_5_beta_release_notes):  _Fixed an issue with incorrect code generation when targeting armv7 devices. (61901594)_

Going to mark this resolved by avoiding Xcode 11.4 with either an earlier or later Xcode version.

If you're seeing the issue on an arm64 device, please open another issue with the details.
